content/scripts/importer.js

```diff
@@ -504,27 +503,19 @@ async function processDocument(
   { usernames, contributors, tags }
 ) {
   const { slug, locale, title, summary } = doc;
-  const localeFolder = path.join(
+
+  const contentPath = path.join(
     isArchive ? archiveRoot : root,
     locale.toLowerCase()
   );
+  const documentPath = Document.buildPath(contentPath, slug);
 
-  const folder = path.join(localeFolder, slugToFoldername(slug));
-  await fs.promises.mkdir(folder, { recursive: true });
-  const htmlFile = path.join(folder, "index.html");
-
-  if (isArchive) {
-    await fs.promises.writeFile(htmlFile, doc.rendered_html);
-  } else {
-    await fs.promises.writeFile(htmlFile, doc.html);
-  }
-
-  const wikiHistoryFile = path.join(folder, "wikihistory.json");
-  const metaFile = path.join(folder, "index.yaml");
-  if (startClean && allBuiltMetaFiles.has(metaFile)) {
-    throw new Error(`${path.resolve(metaFile)} already exists! slug:${slug}`);
+  if (startClean && allBuiltPaths.has(documentPath)) {
```

Comment:
I would've prefered to keep that fully within the document module, but checking whether the path already exists seems to behave differently when done with the slug vs the documentPath 🤔